bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -5293,15 +5293,18 @@ void ReplicaImp::onExecutionFinish() {
   // allowing the create dbCheckpoint operator command to create a dbCheckpoint/snapshot.
   if (getReplicaConfig().dbCheckpointFeatureEnabled && isCurrentPrimary() &&
       DbCheckpointManager::instance().isCreateDbCheckPtSeqNumSet(lastExecutedSeqNum)) {
-    bool noop =
-        (DbCheckpointManager::instance().getNextStableSeqNumToCreateSnapshot().value() != lastExecutedSeqNum + 1);
-    DbCheckpointManager::instance().sendInternalCreateDbCheckpointMsg(
-        lastExecutedSeqNum + 1,
-        noop);  // send noop till we reach stable checkpoint
-    if (!noop) DbCheckpointManager::instance().setNextStableSeqNumToCreateSnapshot(std::nullopt);
-    LOG_INFO(GL, "sendInternalCreateDbCheckpointMsg noop command ");
+    DbCheckpointManager::instance().sendInternalCreateDbCheckpointMsg(lastExecutedSeqNum + 1, true);  // noop=true
+    if (DbCheckpointManager::instance().getNextStableSeqNumToCreateSnapshot().value() == lastExecutedSeqNum + 1) {
```

Comment:
I think we should check if this should be lastExecutedSeqNum + 1 or it should be lastExecutedSeqNum + activeExecutions_, since these many post executions might be happening in parallel.